opcua/ua/uatypes.py

```diff
@@ -163,12 +163,35 @@ class EventNotifier(_MaskEnum):
 
 class Guid(FrozenClass):
 
-    def __init__(self):
-        self.uuid = uuid.uuid4()
+    def __init__(self, value=None):
+        if value is None or value == "":
+            # if no string is supplied, create a new UUID
+            self.uuid = uuid.uuid4()
+        else:
+            try:
+                # attempt to make a UUID with the supplied value, if hex string has errors use a default string
+                self.uuid = uuid.UUID(value)
+            except ValueError as e:
+                # TODO should notify user of error, but this module has no logger, raise UaError stops execution
+                self.uuid = uuid.UUID("00000000-0000-0000-0000-000000000000")
```

Comment:
At lower level it is almost always better to raise exceptions.
you should write something like:
if value is None:
    self.uuid = uuid.uuid4()
elig isinstance(value, uuid): # maybe test size too
    sekf,uuid = val
elif isintance(value, (str, bytes)):
    self.uuid = uuid.UUID(value)
else:
   raise ValueError.....